### PR TITLE
fix(Scripts/Quest): improve Path of Conquest

### PR DESCRIPTION
**Pull request by grimgravy**
## Changes Proposed:
-  fix: now they talk
-  fix: now invocation takes a few seconds
-  fix: now they don't attack

## Issues Addressed:

- Closes https://github.com/chromiecraft/chromiecraft/issues/5060
- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/15171

## Tests Performed:

- 
- 

## How to Test the Changes:

1.
2.
3.

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by FrancescoBorzi** [comment]
@azerothcore/testers @azerothcore/developers can anyone please check this?

**Review by Nefertumm — CHANGES_REQUESTED** [review]

File: src/server/scripts/Outland/zone_shadowmoon_valley.cpp [hunk 1]
```diff
@@ -1635,6 +1635,70 @@ class npc_shadowmoon_tuber_node : public CreatureScript
...
+        Position pos = summoner->GetPosition();
+        me->GetMotionMaster()->MovePoint(POINT_LAND, pos);
+    }
+
+    void MovementInform(uint32 type, uint32 id) override
+    {
+        if (type == POINT_MOTION_TYPE && id == POINT_LAND)
+        {
+            Player* player = ObjectAccessor::GetPlayer(*me, PlayerGUID);
+            Talk(SAY_LAND, player);
```
[the remark this answers is not in the document]
Comment (grimgravy) [reply src/server/scripts/Outland/zone_shadowmoon_valley.cpp:1677]: > Verifique se o player não é nulo antes de usá-lo.

use is mandatory

[the remark this answers is not in the document]
Comment (Kitzunu) [reply src/server/scripts/Outland/zone_shadowmoon_valley.cpp:1677]: He wants you to do
```cpp
if (Player* player = ObjectAccessor::GetPlayer(*me, PlayerGUID))
    Talk(SAY_LAND, player);
```

[the remark this answers is not in the document]
Comment (Kitzunu) [reply src/server/scripts/Outland/zone_shadowmoon_valley.cpp:1677]: To prevent a possible crash

[the remark this answers is not in the document]
Comment (grimgravy) [reply src/server/scripts/Outland/zone_shadowmoon_valley.cpp:1677]: @Kitzunu, sure it could crash?

[the remark this answers is not in the document]
Comment (Nefertumm) [reply src/server/scripts/Outland/zone_shadowmoon_valley.cpp]: Yes, it can

File: data/sql/updates/pending_db_world/rev_1684800494817311100.sql [hunk 2]
```diff
@@ -0,0 +1,19 @@
+--
+DELETE FROM `spell_script_names` WHERE `ScriptName`='spell_calling_korkron_or_wildhammer';
+INSERT INTO `spell_script_names` (`spell_id`, `ScriptName`) VALUES
+(38249, 'spell_calling_korkron_or_wildhammer'),
+(38119, 'spell_calling_korkron_or_wildhammer');
+
+UPDATE `creature_template` SET `ScriptName`='npc_korkron_or_wildhammer' WHERE `entry` IN (22059, 21998);
+
+UPDATE `creature_template` SET `speed_run` = 1.14286 WHERE (`entry` = 21998);
```
Comment (Gultask) [inline data/sql/updates/pending_db_world/rev_1684800494817311100.sql:9]: Speeds are sniffed. Correct value is 2.28571

Comment (grimgravy) [reply data/sql/updates/pending_db_world/rev_1684800494817311100.sql:9]: @Gultask, this figure is high

Comment (Kitzunu) [reply data/sql/updates/pending_db_world/rev_1684800494817311100.sql:9]: It is sniffed, cant be wrong 👍

**Comment by GastonCK** [comment]
Is this ready to test?

**Comment by Kitzunu** [comment]
@Gultask want to take a look?

**Comment by TimekillerTK** [comment]
Tested. 

Behavior exactly as described on `master`:

* The Wildhammer Gryphon Rider/Kor'kron Wind Rider spawn at your position when using the item
* They fight
* They are silent

image

---

On PR `16353`:

* They spawn in the air and take a while to land
* They don't fight
* They have things to say

Footage:

* Horde

https://github.com/azerothcore/azerothcore-wotlk/assets/38417175/3f4b0cab-97bd-4740-b6e9-a0e145489da4

* Alliance

https://github.com/azerothcore/azerothcore-wotlk/assets/38417175/ae21c499-d236-4bc0-ae6c-cf625b481e8a

File: src/server/scripts/Outland/zone_shadowmoon_valley.cpp [hunk 3]
```diff
@@ -1635,6 +1635,71 @@ class npc_shadowmoon_tuber_node : public CreatureScript
...
+        if (type == POINT_MOTION_TYPE && id == POINT_LAND)
+        {
+            if (Player* player = ObjectAccessor::GetPlayer(*me, PlayerGUID))
+                Talk(SAY_LAND, player);
+
+            me->RemoveFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_NOT_SELECTABLE);
+        }
+    }
+private:
+    ObjectGuid PlayerGUID;
```
Comment (Nefertumm) [inline src/server/scripts/Outland/zone_shadowmoon_valley.cpp:1683]:
```suggestion
    ObjectGuid _playerGUID;
```

File: src/server/scripts/Outland/zone_shadowmoon_valley.cpp [hunk 4]
```diff
@@ -1635,6 +1635,71 @@ class npc_shadowmoon_tuber_node : public CreatureScript
...
+        me->SetFacingToObject(summoner);
+        Position pos = summoner->GetPosition();
+        me->GetMotionMaster()->MovePoint(POINT_LAND, pos);
+    }
+
+    void MovementInform(uint32 type, uint32 id) override
+    {
+        if (type == POINT_MOTION_TYPE && id == POINT_LAND)
+        {
+            if (Player* player = ObjectAccessor::GetPlayer(*me, PlayerGUID))
```
Comment (Nefertumm) [inline src/server/scripts/Outland/zone_shadowmoon_valley.cpp:1676]:
```suggestion
            if (Player* player = ObjectAccessor::GetPlayer(*me, _playerGUID))
```

File: src/server/scripts/Outland/zone_shadowmoon_valley.cpp [hunk 5]
```diff
@@ -1635,6 +1635,71 @@ class npc_shadowmoon_tuber_node : public CreatureScript
...
+    }
+
+    void JustDied(Unit* /*killer*/) override
+    {
+        me->DespawnOrUnsummon(3s, 0s);
+    }
+
+    void IsSummonedBy(WorldObject* summoner) override
+    {
+        PlayerGUID = summoner->GetGUID();
```
Comment (Nefertumm) [inline src/server/scripts/Outland/zone_shadowmoon_valley.cpp:1666]:
```suggestion
        _playerGUID = summoner->GetGUID();
```

File: data/sql/updates/pending_db_world/rev_1684800494817311100.sql [hunk 6]
```diff
@@ -0,0 +1,19 @@
+--
+DELETE FROM `spell_script_names` WHERE `ScriptName`='spell_calling_korkron_or_wildhammer';
+INSERT INTO `spell_script_names` (`spell_id`, `ScriptName`) VALUES
+(38249, 'spell_calling_korkron_or_wildhammer'),
+(38119, 'spell_calling_korkron_or_wildhammer');
+
+UPDATE `creature_template` SET `ScriptName`='npc_korkron_or_wildhammer' WHERE `entry` IN (22059, 21998);
+
+UPDATE `creature_template` SET `speed_run` = 2.28571 WHERE `entry` IN (22059, 21998);
```
Comment (Nefertumm) [inline data/sql/updates/pending_db_world/rev_1684800494817311100.sql:9]: source?

File: src/server/scripts/Outland/zone_shadowmoon_valley.cpp (hunk 4 above)
Comment (grimgravy) [reply src/server/scripts/Outland/zone_shadowmoon_valley.cpp:1676]: ?

Comment (Nefertumm) [reply src/server/scripts/Outland/zone_shadowmoon_valley.cpp:1676]: Change the name of that variable to be properly identified as private, and along with that change the P to p as capitalized names are used only for naming classes and not variables.

**Review by Nefertumm — APPROVED** [review]

**Pull request merged**